common/src/test/java/com/microsoft/identity/common/BrowserSelectorTest.java

```diff
@@ -253,14 +225,27 @@ public TestBrowserBuilder setVersion(String version) {
             return this;
         }
 
+        private PackageInfo addSignatures(final PackageInfo packageInfo, final Signature[] signatures) {
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P) {
+                packageInfo.signatures = signatures;
+                return packageInfo;
+            }
+
+            final SigningInfo signingInfo = mock(SigningInfo.class);
+            when(signingInfo.hasMultipleSigners()).thenReturn(false);
+            when(signingInfo.getSigningCertificateHistory()).thenReturn(signatures);
+            packageInfo.signingInfo = signingInfo;
```

Comment:
Can this be done with Robolectric too?

Reply:
Don't know that the methodology matters, but when you're writing code, if you want to make it easy to test - implement an interface, and provide a way to construct your POJOs.

Also, at some point, SigningInfo has been made final.

Reply:
Spent a few mins with it and couldn't think of a way of doing this cleanly with robolectric (this is 'per browser', and there are >1 browsers in each tests), so I'm keeping this as is for now.